src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/events/InboundEventsListener.kt

```diff
@@ -0,0 +1,53 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service.events
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.PropertyNamingStrategies
+import com.fasterxml.jackson.databind.annotation.JsonNaming
+import com.fasterxml.jackson.module.kotlin.readValue
+import io.awspring.cloud.sqs.annotation.SqsListener
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+@Component
+class InboundEventsListener(
+  private val mapper: ObjectMapper,
+  private val inboundEventsService: InboundEventsService,
+) {
+  companion object {
+    private val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  init {
+    log.info("Event listener started.")
+  }
+
+  @SqsListener("education-and-work-plan", factory = "hmppsQueueContainerFactoryProxy")
+  internal fun onMessage(rawMessage: String) {
+    log.info("Inbound event raw message: $rawMessage")
+
+    val sqsMessage: SQSMessage = mapper.readValue(rawMessage)
+
+    when (sqsMessage.Type) {
+      "Notification" -> {
```

Comment:
What if the type is not `Notification` .... ?
Where does this value get set? 
Can we use a const?

Reply:
It's all done my the domain-events framework. Have a look at that _"NDSS-ADR - Domain Event Schema"_ PDF which Catriona sent.